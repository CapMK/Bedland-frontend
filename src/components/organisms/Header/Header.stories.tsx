import { PARAM_REDUX_MERGE_STATE } from 'addon-redux';
import type { Meta, StoryObj } from '@storybook/react';
import { BrowserRouter } from 'react-router-dom';
import Header from './Header';

const meta = {
  title: 'Organisms/Header',
  component: Header,
  tags: ['autodocs'],
  decorators: [(Story) => <BrowserRouter>{Story()}</BrowserRouter>],
} satisfies Meta<typeof Header>;

export default meta;
type Story = StoryObj<typeof meta>;

export const ResidentLoginPage: Story = {
  parameters: {
    [PARAM_REDUX_MERGE_STATE]: '{"user": {"role": "resident"}}',
  },
};
export const ManagerLoginPage: Story = {
  parameters: {
    [PARAM_REDUX_MERGE_STATE]: '{"user": {"role": "manager"}}',
  },
};

export const ManagerDashboardPage: Story = {
  parameters: {
    [PARAM_REDUX_MERGE_STATE]: '{"user": {"role": "manager", "loggedIn": "true"}}',
  },
};
export const ResidentDashboardPage: Story = {
  parameters: {
    [PARAM_REDUX_MERGE_STATE]: '{"user": {"role": "resident", "loggedIn": "true"}}',
  },
};
