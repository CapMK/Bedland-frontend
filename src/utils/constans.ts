export const HEADER_HEIGHT = 70;
export const FOOTER_HEIGHT = 52;
export const LOGIN_INPUT_WIDTH = 19.875;

// TODO
// BASE_URL points to json server
// Replace with actual API url when BE is ready
export const BASE_URL = 'http://localhost:3001';
