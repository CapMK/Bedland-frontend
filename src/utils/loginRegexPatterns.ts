import { InputVariant, TRegister } from 'components/atoms/Input/Input.types';
import { Getters } from 'types/shared.types';

type LoginRegexProps = Getters<InputVariant, TRegister>;

const loginRegexPattern: LoginRegexProps = {
  login: {
    pattern: /^[a-zA-Z0-9]+$/,
    required: true,
    maxLength: 36,
    minLength: 8,
  },
  password: {
    pattern: /^(?=.*\d)(?=.*[a-z])(?=.*[A-Z])(?=.*[!@#$%^&*(),<.>/?;:'"]).{8,24}$/,
    required: true,
    maxLength: 36,
    minLength: 8,
  },
  confirmPassword: {
    required: true,
  },
};

export default loginRegexPattern;
// ^(?=.*[A-Za-z])(?=.*\d)(?=.*[@$!%*#?&])[A-Za-z\d@$!%*#?&]{8,}$ <- regex for password that should work.
// /^(?=.*[A-Za-z])(?=.*\d)[A-Za-z\d]{8,}$/,
