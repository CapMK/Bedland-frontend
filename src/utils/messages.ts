const ERROR_MESSAGES = {
  resetPasswordPage: {
    notWithPasswordsRules:
      'Password is incorrect, please try again after reviewing the rules above',

    passwordsDoesNotMatchUp: 'Passwords do not match to each other',
  },
  loginPage: {
    errorMessage: 'Your login or password is incorrect.',
  },
};

export default ERROR_MESSAGES;
