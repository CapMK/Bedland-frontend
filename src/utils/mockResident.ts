const mockResidentList = [
  {
    id: 1,
    version: 0,
    createDate: '2023-02-14T00:00:00',
    updateDate: '2023-02-14T00:00:00',
    login: 'iSutherland',
    password: 'IsABEl',
    name: 'Isabel',
    lastName: 'Sutherland',
    email: 'isabel.sutherland@email.com',
    phoneNumber: '123456789',
    flatId: 1,
    owner: true,
  },
  {
    id: 2,
    version: 0,
    createDate: '2023-02-14T00:00:00',
    updateDate: '2023-02-14T00:00:00',
    login: 'clWeiss',
    password: 'ClWeiss',
    name: 'Claude',
    lastName: 'Weiss',
    email: 'Claude.Weiss@email.com',
    phoneNumber: '123456789',
    flatId: 2,
    owner: false,
  },
];

export default mockResidentList;
