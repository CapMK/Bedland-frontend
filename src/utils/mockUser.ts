import { User } from 'store/reducers/user/user_slice';

export const mockUserCredentials = {
  residentUsername: 'testmember',
  managerUsername: 'testmanager',
  password: '!Test123!',
};

export const loggedInManagerUser = {
  loggedIn: true,
  status: 'success',
  error: undefined,
  userId: '122',
  login: mockUserCredentials.managerUsername,
  token: undefined,
  role: 'manager',
} as User;

export const loggedInResidentUser = {
  loggedIn: true,
  status: 'success',
  error: undefined,
  userId: '122',
  login: mockUserCredentials.residentUsername,
  token: undefined,
  role: 'resident',
} as User;

export const loggedOutManagerUser = {
  loggedIn: false,
  status: 'idle',
  userId: undefined,
  login: undefined,
  token: undefined,
  role: 'manager',
} as User;

export const loggedOutResidentUser = {
  loggedIn: false,
  status: 'idle',
  userId: undefined,
  login: undefined,
  token: undefined,
  role: 'resident',
} as User;
