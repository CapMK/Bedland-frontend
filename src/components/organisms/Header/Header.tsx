import routes from 'App/routing/routes';
import LogoManager from 'assets/img/LogoManager.svg';
import LogoResident from 'assets/img/LogoResident.svg';
import AccountDropdown from 'components/organisms/AccountDropdown/AccountDropdown';
import { useContext } from 'react';
import { useLocation } from 'react-router-dom';
import { useAppSelector } from 'store/hooks';
import { selectUserIsLoggedIn } from 'store/reducers/user/user_slice';
import { ThemeContext } from 'theme/ThemeContext';
import * as Styled from './Header.styled';

const Header = () => {
  const isLoggedIn: boolean = useAppSelector(selectUserIsLoggedIn);
  const location = useLocation();
  const userRole = useAppSelector((state) => state.user.role);
  const { palette } = useContext(ThemeContext);

  return (
    <nav>
      <Styled.HeaderBox isLoggedIn={isLoggedIn} palette={palette}>
        <img
          src={
            userRole === 'manager' || location.pathname === routes.homeForNotLoggedInManager
              ? LogoManager
              : LogoResident
          }
          alt='Logo Bedland'
        />
        {isLoggedIn ? <AccountDropdown /> : null}
      </Styled.HeaderBox>
    </nav>
  );
};

export default Header;
