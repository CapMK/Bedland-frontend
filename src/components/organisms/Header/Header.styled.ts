import styled from 'styled-components';
import { HEADER_HEIGHT } from 'utils/constans';
import { PaletteType } from 'theme/theme.types';

type HeaderProps = {
  isLoggedIn: boolean;
  palette: PaletteType;
};

export const HeaderBox = styled.div<HeaderProps>`
  align-items: center;
  background-color: ${({ palette }) => palette.light};
  display: flex;
  gap: 10px;
  height: ${HEADER_HEIGHT}px;
  justify-content: ${({ isLoggedIn }) => (!isLoggedIn ? 'center' : 'space-between')};
  position: relative;
  width: 100%;
  z-index: 3;
`;

export default HeaderBox;
