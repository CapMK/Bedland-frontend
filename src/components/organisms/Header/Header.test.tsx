import { cleanup, screen } from '@testing-library/react';
import { loggedInManagerUser, loggedInResidentUser } from 'utils/mockUser';
import { renderWithProviders } from 'utils/test.utils';
import Header from './Header';

describe('Header', () => {
  afterEach(cleanup);
  it('headers renders correctly for logged out user', () => {
    const snapshot = renderWithProviders(<Header />, {});

    expect(snapshot).toMatchSnapshot();
  });
  it('headers renders without dropdown for logged out user', () => {
    renderWithProviders(<Header />, {});
    const dropdown = screen.queryByTestId('dropdown');

    expect(dropdown).toBeFalsy();
  });
  it('header shows dropdown, if a user is logged in ', () => {
    renderWithProviders(<Header />, {
      preloadedState: {
        user: loggedInManagerUser,
      },
    });
    const dropdown = screen.queryByTestId('dropdown');
    expect(dropdown).toBeTruthy();
  });

  it('headers renders correctly for logged in manager', () => {
    const managerHeader = () =>
      renderWithProviders(<Header />, {
        preloadedState: {
          user: loggedInManagerUser,
        },
      });

    expect(managerHeader()).toMatchSnapshot();
  });

  it('headers renders correctly for logged in resident', () => {
    const managerHeader = () =>
      renderWithProviders(<Header />, {
        preloadedState: {
          user: loggedInResidentUser,
        },
      });

    expect(managerHeader()).toMatchSnapshot();
  });
});
