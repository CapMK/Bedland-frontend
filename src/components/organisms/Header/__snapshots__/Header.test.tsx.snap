// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header > headers renders correctly for logged in manager 1`] = `
{
  "asFragment": [Function],
  "baseElement": .c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c3 {
  border-radius: 50%;
  height: 2rem;
  width: 2rem;
}

.c1 {
  display: block;
  height: 100%;
  position: relative;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  gap: 1rem;
  height: 70px;
  margin-right: 2rem;
  min-width: 11.5rem;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #FCFEFF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 10px;
  height: 70px;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  position: relative;
  width: 100%;
  z-index: 3;
}

<body>
    <div>
      <nav>
        <div
          class="c0"
        >
          <img
            alt="Logo Bedland"
            src="/src/assets/img/LogoManager.svg"
          />
          <div
            class="c1"
          >
            <div
              class="c2"
              data-testid="dropdown"
            >
              <img
                alt="profile"
                class="c3"
                size="2"
                src="/src/assets/img/Avatar.svg"
              />
              <div
                class="c4"
              >
                <h5
                  style="font-family: Source Sans Pro, sans-serif; font-weight: 600; font-size: 0.875rem; line-height: 1.4; letter-spacing: 0; color: rgb(27, 26, 26);"
                />
                <h5
                  style="font-family: Source Sans Pro, sans-serif; font-weight: 400; font-size: 0.75rem; line-height: 1.4; letter-spacing: 0; color: rgb(75, 75, 75);"
                >
                  manager
                </h5>
              </div>
              <svg
                data-testid="more-icon"
                fill="none"
                height="20"
                viewBox="0 0 20 20"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <mask
                  id="mask-more"
                />
                <path
                  d="M10 9.99251V10.0075M10 4V4.01498M10 15.985V16"
                  stroke="#1B1A1A"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2.68161"
                />
              </svg>
            </div>
          </div>
        </div>
      </nav>
      <div
        data-testid="location-display"
      >
        /
      </div>
    </div>
  </body>,
  "container": .c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c3 {
  border-radius: 50%;
  height: 2rem;
  width: 2rem;
}

.c1 {
  display: block;
  height: 100%;
  position: relative;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  gap: 1rem;
  height: 70px;
  margin-right: 2rem;
  min-width: 11.5rem;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #FCFEFF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 10px;
  height: 70px;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  position: relative;
  width: 100%;
  z-index: 3;
}

<div>
    <nav>
      <div
        class="c0"
      >
        <img
          alt="Logo Bedland"
          src="/src/assets/img/LogoManager.svg"
        />
        <div
          class="c1"
        >
          <div
            class="c2"
            data-testid="dropdown"
          >
            <img
              alt="profile"
              class="c3"
              size="2"
              src="/src/assets/img/Avatar.svg"
            />
            <div
              class="c4"
            >
              <h5
                style="font-family: Source Sans Pro, sans-serif; font-weight: 600; font-size: 0.875rem; line-height: 1.4; letter-spacing: 0; color: rgb(27, 26, 26);"
              />
              <h5
                style="font-family: Source Sans Pro, sans-serif; font-weight: 400; font-size: 0.75rem; line-height: 1.4; letter-spacing: 0; color: rgb(75, 75, 75);"
              >
                manager
              </h5>
            </div>
            <svg
              data-testid="more-icon"
              fill="none"
              height="20"
              viewBox="0 0 20 20"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <mask
                id="mask-more"
              />
              <path
                d="M10 9.99251V10.0075M10 4V4.01498M10 15.985V16"
                stroke="#1B1A1A"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2.68161"
              />
            </svg>
          </div>
        </div>
      </div>
    </nav>
    <div
      data-testid="location-display"
    >
      /
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;

exports[`Header > headers renders correctly for logged in resident 1`] = `
{
  "asFragment": [Function],
  "baseElement": .c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c3 {
  border-radius: 50%;
  height: 2rem;
  width: 2rem;
}

.c1 {
  display: block;
  height: 100%;
  position: relative;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  gap: 1rem;
  height: 70px;
  margin-right: 2rem;
  min-width: 11.5rem;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #FCFEFF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 10px;
  height: 70px;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  position: relative;
  width: 100%;
  z-index: 3;
}

<body>
    <div>
      <nav>
        <div
          class="c0"
        >
          <img
            alt="Logo Bedland"
            src="/src/assets/img/LogoResident.svg"
          />
          <div
            class="c1"
          >
            <div
              class="c2"
              data-testid="dropdown"
            >
              <img
                alt="profile"
                class="c3"
                size="2"
                src="/src/assets/img/Avatar.svg"
              />
              <div
                class="c4"
              >
                <h5
                  style="font-family: Source Sans Pro, sans-serif; font-weight: 600; font-size: 0.875rem; line-height: 1.4; letter-spacing: 0; color: rgb(27, 26, 26);"
                />
                <h5
                  style="font-family: Source Sans Pro, sans-serif; font-weight: 400; font-size: 0.75rem; line-height: 1.4; letter-spacing: 0; color: rgb(75, 75, 75);"
                >
                  resident
                </h5>
              </div>
              <svg
                data-testid="more-icon"
                fill="none"
                height="20"
                viewBox="0 0 20 20"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <mask
                  id="mask-more"
                />
                <path
                  d="M10 9.99251V10.0075M10 4V4.01498M10 15.985V16"
                  stroke="#1B1A1A"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2.68161"
                />
              </svg>
            </div>
          </div>
        </div>
      </nav>
      <div
        data-testid="location-display"
      >
        /
      </div>
    </div>
  </body>,
  "container": .c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c3 {
  border-radius: 50%;
  height: 2rem;
  width: 2rem;
}

.c1 {
  display: block;
  height: 100%;
  position: relative;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  gap: 1rem;
  height: 70px;
  margin-right: 2rem;
  min-width: 11.5rem;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #FCFEFF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 10px;
  height: 70px;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  position: relative;
  width: 100%;
  z-index: 3;
}

<div>
    <nav>
      <div
        class="c0"
      >
        <img
          alt="Logo Bedland"
          src="/src/assets/img/LogoResident.svg"
        />
        <div
          class="c1"
        >
          <div
            class="c2"
            data-testid="dropdown"
          >
            <img
              alt="profile"
              class="c3"
              size="2"
              src="/src/assets/img/Avatar.svg"
            />
            <div
              class="c4"
            >
              <h5
                style="font-family: Source Sans Pro, sans-serif; font-weight: 600; font-size: 0.875rem; line-height: 1.4; letter-spacing: 0; color: rgb(27, 26, 26);"
              />
              <h5
                style="font-family: Source Sans Pro, sans-serif; font-weight: 400; font-size: 0.75rem; line-height: 1.4; letter-spacing: 0; color: rgb(75, 75, 75);"
              >
                resident
              </h5>
            </div>
            <svg
              data-testid="more-icon"
              fill="none"
              height="20"
              viewBox="0 0 20 20"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <mask
                id="mask-more"
              />
              <path
                d="M10 9.99251V10.0075M10 4V4.01498M10 15.985V16"
                stroke="#1B1A1A"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2.68161"
              />
            </svg>
          </div>
        </div>
      </div>
    </nav>
    <div
      data-testid="location-display"
    >
      /
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;

exports[`Header > headers renders correctly for logged out user 1`] = `
{
  "asFragment": [Function],
  "baseElement": .c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #FCFEFF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 10px;
  height: 70px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
  width: 100%;
  z-index: 3;
}

<body>
    <div>
      <nav>
        <div
          class="c0"
        >
          <img
            alt="Logo Bedland"
            src="/src/assets/img/LogoResident.svg"
          />
        </div>
      </nav>
      <div
        data-testid="location-display"
      >
        /
      </div>
    </div>
  </body>,
  "container": .c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #FCFEFF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 10px;
  height: 70px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
  width: 100%;
  z-index: 3;
}

<div>
    <nav>
      <div
        class="c0"
      >
        <img
          alt="Logo Bedland"
          src="/src/assets/img/LogoResident.svg"
        />
      </div>
    </nav>
    <div
      data-testid="location-display"
    >
      /
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
